backend/src/test/groovy/com/capstone/moneytree/service/DefaultTransactionServiceTest.groovy

```diff
@@ -28,110 +35,131 @@ import spock.lang.Specification
-
-      and: "user dao returns a user"
-      def user = new User()
-      userDao.findUserById(_ as Long) >> user
-
-      and: "returns a balance"
-      def balance = "1234.56"
-      api.getAccount() >> Stub(Account) {
-         getCash() >> balance
-      }
-
-      when: "execute is triggered"
-      transactionService.execute("123456789", order)
-
-      then:
-      1 * userDao.save(_)
-      user.getTransactions().size() == 1
-      user.getBalance() == Float.parseFloat(balance)
-      user.getTransactions()[0].with {
-         it.getStatus() == TransactionStatus.PENDING
-         it.getMoneyTreeOrderType() == MoneyTreeOrderType.MARKET_BUY
-      }
-   }
-
-   def "Should throw an exception when the userId does not exist"() {
-      given: "Database returns null (no user for that id)"
-      userDao.findUserById(123456789) >> null
-
-      when: "executing a transaction for a non existing user"
-      transactionService.execute("1234", null)
-
-      then:
-      thrown(EntityNotFoundException)
-   }
-
-   def "exception thrown by api when making call should be caught"() {
-      given: "The session returns the mock api"
-      session.alpaca(_) >> api
-
-      and: "a valid order"
-      def order = buildMarketOrder()
-
-      and: "user dao returns a user"
-      def user = new User()
-      userDao.findUserById(_ as Long) >> user
-
-      and: "api request throws exception"
-      api.requestNewMarketOrder(_, _, _, _) >> {
-         throw new Exception("for test")
-      }
-
-      when:
-      transactionService.execute("12345678", order)
-
-      then:
-      thrown(AlpacaException)
-   }
-
-   Stock buildStock(Asset asset) {
-      return Stock.builder()
-              .status(asset.getStatus())
-              .symbol(asset.getSymbol())
-              .exchange(asset.getExchange())
-              .build()
-   }
+
+    TransactionDao transactionDao = Mock()
+    UserDao userDao = Mock()
+    StockDao stockDao = Mock();
+    MadeDao madeDao = Mock();
+    ToFulfillDao toFulfillDao = Mock();
+    AlpacaSession session = Mock()
+    AlpacaAPI api = Mock()
+    StockMarketDataService stockMarketDataService = Mock()
+    TransactionService transactionService = new DefaultTransactionService(transactionDao, userDao, stockDao, madeDao, toFulfillDao, session, stockMarketDataService,)
+
+    def "Validate that findAll calls the transactionDao"() {
+        when:
+        transactionService.getAllTransactions()
+
+        then:
+        1 * transactionDao.findAll()
+    }
+
+    def "Validate that findTransactionsByOrderType filters correctly"() {
+        given: "A list of transaction with different order type"
+        transactionDao.findByMoneyTreeOrderType(MoneyTreeOrderType.MARKET_BUY) >> [buildTransactions()[0], buildTransactions()[1], buildTransactions()[2]]
+
+        when: "We get all the market buy type of transactions"
+        def marketBuyTransactions = transactionService.getTransactionsByOrderType(MoneyTreeOrderType.MARKET_BUY)
```

Comment:
Hmm we may have to write tests for MARKET_SELL as well. Not sure what`s the difference in terms of API so I'm curious to know if it is the case.

Reply:
for this story we only had MARKET_BUY but yes definitely we need to have tests for other types as well.